android/src/main/java/expo/modules/alternateappicons/ExpoAlternateAppIconsModule.kt

```diff
@@ -24,21 +23,19 @@ class ExpoAlternateAppIconsModule : Module() {
   }
 
   private fun getAppIconName(): String? {
-    val activityName = appContext.activityProvider?.currentActivity?.componentName?.shortClassName
+    val currentActivityComponent = appContext.activityProvider!!.currentActivity!!.componentName!!
```

Comment:
Is this assertion safe?

Reply:
This assertion throws an exception if we try to retrieve the activity before it's available.
When this happens, you probably prefer to receive an exception react-side, instead of a null value.